compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1345,7 +1345,7 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
     }
 
     def isScalaPhantomSelf(self: tpd.ValDef): Boolean = {
-      if (self.tpt.tpe.derivesFrom(defn.PhantomClass)) {
+      if (self.tpt.tpe.classSymbol eq defn.PhantomClass) {
```

Comment:
FYI, e82f1ebccef851b727828095d96628dfef951a9a was the fix to the issue that I wanted to avoid.